pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -89,8 +96,14 @@ func (r *DNSPolicyReconciler) reconcileGatewayHealthChecks(ctx context.Context,
 	return allResults, nil
 }
 
-func (r *DNSPolicyReconciler) reconcileDNSRecordHealthChecks(ctx context.Context, dnsRecord *v1alpha1.DNSRecord, config *healthChecksConfig) ([]dns.HealthCheckResult, error) {
-	healthCheckReconciler := r.DNSProvider.HealthCheckReconciler()
+func (r *DNSPolicyReconciler) reconcileDNSRecordHealthChecks(ctx context.Context, dnsRecord *v1alpha1.DNSRecord, config *healthChecksConfig, managedzone *v1alpha1.ManagedZone) ([]dns.HealthCheckResult, error) {
+
+	DNSProvider, err := r.DNSProv.CreateDNSProvider(ctx, managedzone)
```

Comment:
maybe I am missing something here? Not sure why we are creating a DNSProvider when reconciling a health check? I would have thought we would fail to reconcile health checks until there was a DNSProvider present. Same with managed zones and DNSRecords. If there is no provider present we can't really do anything

Reply:
nvm I see it is actually not creating the DNSProvider resource